src/control/tools/strokestabilizer/ArithmeticStrokeStabilizer.cpp

```diff
@@ -0,0 +1,70 @@
+#include "ArithmeticStrokeStabilizer.h"
+
+#include <cmath>
+#include <numeric>
+
+// #include <control/XournalMain.h>
+// #include <gdk/gdk.h>
+
+#include "control/tools/StrokeHandler.h"  // TODO Ugly fix, remove!
+
+// Needed?
+// #include "control/settings/Settings.h"
+// #include "config-features.h"
+
+ArithmeticStrokeStabilizer::ArithmeticStrokeStabilizer(const PositionInputData& pos):
+        // TODO read config
+        bufferLength(std::max(2, StrokeHandler::stabilizingBufferSize)) {
+    /**
+     * Fill the queue with copies of our starting point
+     * Is this the best way to do this? We could also aggregate the first events as they come
+     * but this would add a test (on the length of the queue) at every event triggered.
+     */
+    for (int i = 1; i < bufferLength; i++) {
+        // The next event will be pushed before averaging, so there will be exactly bufferLength elements
+        // in the queue then.
+        eventBuffer.emplace_front(pos.x, pos.y, pos.pressure);
+    }
+    g_message("Created ArithmeticStrokeStabilizer with bufferLength = %d", bufferLength);
+}
+
+void ArithmeticStrokeStabilizer::stabilizePointUsingEvent(const PositionInputData& pos, double zoom, Point* point) {
+
+    /**
+     * Create and push the event
+     */
+    eventBuffer.emplace_front(pos.x, pos.y, pos.pressure);
+
+    /**
+     * Average the coordinates using an arithmetic mean
+     */
+    BufferedEvent sum =
+            std::reduce(begin(eventBuffer), end(eventBuffer), BufferedEvent(0.0, 0.0, 0.0), [](auto&& lhs, auto&& rhs) {
+                return BufferedEvent(lhs.x + rhs.x, lhs.y + rhs.y, lhs.pressure + rhs.pressure);
+            });
```

Comment:
Hmm... It turns the CI Test_Build for Linux does not know about std::reduce. It definitely compiles fine on my machine. Any idea as to why?

Reply:
From the log, it looks like the Linux build is using `g++8` with flags that include `c++17`:
```sh
2020-12-11T19:22:33.2615758Z FAILED: src/CMakeFiles/xournalpp-core.dir/control/tools/strokestabilizer/ArithmeticStrokeStabilizer.cpp.o 
2020-12-11T19:22:33.2619953Z /usr/bin/g++-8 -DGDK_VERSION_MIN_REQUIRED=GDK_VERSION_3_18 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_32 -D_USE_MATH_DEFINES -Isrc -I../src -I../src/util -isystem /usr/include/glib-2.0 -isystem /usr/lib/x86_64-linux-gnu/glib-2.0/include -isystem /usr/include/gtk-3.0 -isystem /usr/include/at-spi2-atk/2.0 -isystem /usr/include/at-spi-2.0 -isystem /usr/include/dbus-1.0 -isystem /usr/lib/x86_64-linux-gnu/dbus-1.0/include -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/cairo -isystem /usr/include/pango-1.0 -isystem /usr/include/harfbuzz -isystem /usr/include/atk-1.0 -isystem /usr/include/pixman-1 -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/libxml2 -isystem /usr/include/poppler/glib -isystem /usr/include/poppler -isystem /usr/include/lua5.3 -g   -g -Wreturn-type -Wuninitialized -Wunused-value -Wunused-variable -Wall -std=c++17 -MD -MT src/CMakeFiles/xournalpp-core.dir/control/tools/strokestabilizer/ArithmeticStrokeStabilizer.cpp.o -MF src/CMakeFiles/xournalpp-core.dir/control/tools/strokestabilizer/ArithmeticStrokeStabilizer.cpp.o.d -o src/CMakeFiles/xournalpp-core.dir/control/tools/strokestabilizer/ArithmeticStrokeStabilizer.cpp.o -c ../src/control/tools/strokestabilizer/ArithmeticStrokeStabilizer.cpp
```

From [this](https://stackoverflow.com/questions/43689444/header-execution-and-stdreduce-not-found), it looks like it could be the version of `gcc`, but I'm not sure...

Regardless, I hope this is helpful!

Reply:
Thanks! Yes, anything I can find on the matter refers to the version of g++. I don't think it's worth spending time on here. We could simply not use std::reduce here.